core/src/main/scala/akka/persistence/cassandra/query/EventsByTagStage.scala

```diff
@@ -563,6 +697,34 @@ import scala.compat.java8.FutureConverters._
         }
       }
 
+      private def setLookingForMissingState(
+          events: List[UUIDRow],
+          fromOffset: UUID,
+          toOffset: UUID,
+          missingOffsets: Map[PersistenceId, MissingData],
+          missingSequenceNrs: Map[PersistenceId, Set[Long]],
+          explicitGapDetected: Boolean,
+          timeout: FiniteDuration): Unit = {
+        // Start in the toOffsetBucket as it is considered more likely an event has been missed due to an event
+        // being delayed slightly rather than by a full bucket
+        val bucket = TimeBucket(toOffset, bucketSize)
+        // Could the event be in the previous bucket?
+        val lookInPrevious = !bucket.within(toOffset)
```

Comment:
shouldn't this be if `fromOffset` isn't also in the `toOffset` bucket or am I misunderstanding things?

Reply:
You're right. It still works as that flag is reset every time (on line 819) but this means we'll take an extra `refresh-period` before we query the previous bucket. The tests were lenient enough with their timings that they passed with this extra `refrsh-interval` i'll tighten one up and hope it doesn't get flaky in travis.

Reply:
Nice spot!

Reply:
Added a test in 48df46cf82a6c14de5e57caae2f3cc9e6b77efcb that fails, had to add a new spec to raise the refresh-interval otherwise it slows down the rest of the test too much